Add compatibility with Pyside2/Qt5

amotl opened the pull request:
Dear Uwe, Jens and all sigrok contributors,

first things first: Thanks a stack for conceiving and maintaining sigrok.

While we know PulseView and SmuView are in the focus of development, there might still be a need for `sigrok-meter`. A friend told me that he would be interested to run `sigrok-meter` on Python 3. So, I tried to look into that and after finding it would still be based on PySide/Qt, I decided to give it a shot to check what it would take to modernize it to use PySide2/Qt5. Turns out, not much!

So, while I don't know if this is an accepted method to hand in patches to the code base, I still wanted to publish it here. Let me know what you think about it and also if you have a different process of submitting patches.

Keep up the spirit and with kind regards,
Andreas.

abraxa commented:
Thanks a lot for the changes!

In the .rst file you refer to "git clone https://github.com/daq-tools/sigrok-meter --branch pyside2", which as you know isn't the main repo. From a quick glance, it seems as if it merely contains the changes of this PR, so if the PR is merged, the .rst file might as well refer to the sigrok-meter main repo.

Is my understanding correct?

amotl commented:
Hi Soeren,

> `--branch pyside2` - ... so if the PR is merged, the .rst file might as well refer to the sigrok-meter main repo.

Thanks for spotting this flaw. I merely added this to provide my colleague with thorough installation instructions and forgot to remove it before submitting the patch.

I've now corrected this and also improved the `INSTALL.rst` to provide better guidance while being at it.

With kind regards,
Andreas.

amotl commented:
Hi Soeren,

I've addressed your suggestions the other day. Please let me know if you see that any other amendments should be made.

With kind regards,
Andreas.

amotl commented:
Dear Soeren,

is there a chance to merge this patch?

With kind regards,
Andreas.

fenugrec commented:
FYI it is being discussed on the sigrok IRC channel these days if you would still care to participate. The PR is not mergeable as-is, there is some interest, but some broader questions need to be addressed first.

dragonmux commented on .gitignore:
```diff
@@ -1,2 +1,3 @@
+.idea
```
This seems unrelated to the main change set - perhaps drop this change?

dragonmux commented on Makefile:
```diff
@@ -24,13 +24,21 @@ ifneq ($(MAKECMDGOALS),clean)
       RCC = pyside-rcc
     else
-      $(error "resource compiler not found")
+      ifeq (0,$(shell which pyside2-rcc >/dev/null 2>&1; echo $$?))
```
Is there a particular reason for checking for PySide2's resource compiler /after/ PySide's? Would have thought you'd want this the other way around. Additionally, the Python changes drop support for PySide, so what point is there keeping the old build logic around?

dragonmux commented on acquisition.py:
```diff
@@ -98,14 +99,14 @@ def add_device(self, driverstring, configstring):
 
         device = devs[0]
 
+        self.session.add_device(device)
+        device.open()
```
What is the purpose of moving these two lines? This reads like change noise - a comment inserted before on why this reordering is needed and what bug it solves would be very helpful.

dragonmux commented on acquisition.py:
```diff
@@ -114,6 +115,7 @@ def is_running(self):
     def start(self):
         '''Start the session.'''
         self.session.start()
+        self.session.run()
```
This looks like a behaviour change, why is this necessary?

dragonmux reviewed (commented):
Having had a conversation with the sigrok folks on IRC, we've had a look through and identified some items that could do with clarification and potentially clean-up.

It would also be helpful if commit messages like "Make the data flow" were improved to better explain what the commits do and summarise. Right now, the commit log messages don't make much sense to us.
